ee/hogai/session_recordings_filters/nodes.py

```diff
@@ -0,0 +1,67 @@
+from uuid import uuid4
+
+from langchain_core.prompts import ChatPromptTemplate
+from langchain_core.runnables import RunnableConfig
+from langchain_openai import ChatOpenAI
+
+from ee.hogai.session_recordings_filters.prompts import (
+    AI_FILTER_INITIAL_PROMPT,
+    AI_FILTER_PROPERTIES_PROMPT,
+    AI_FILTER_REQUEST_PROMPT,
+)
+from ee.hogai.utils.nodes import AssistantNode
+from ee.hogai.utils.types import AssistantState, PartialAssistantState
+from posthog.schema import AssistantContextualTool, AssistantToolCallMessage, MaxRecordingUniversalFilters
+
+
+class SessionRecordingsFiltersNode(AssistantNode):
+    """
+    Node for generating session recording filters using AI.
+
+    This node takes a user query about session recordings and generates
+    structured filters that can be applied to the list of recordings.
+    """
+
+    def run(self, state: AssistantState, config: RunnableConfig) -> PartialAssistantState:
+        assert state.root_tool_call_id is not None
+        tool_call = self._get_tool_call(state.messages, state.root_tool_call_id)
+
+        model = (
+            ChatOpenAI(model="gpt-4o", temperature=0.2)
+            .with_structured_output(MaxRecordingUniversalFilters, include_raw=False)
```

Comment:
Structured output by default uses function calling, which doesn't enforce a strict schema. Since the schema is a Pydantic object and we expect a structured response, we should enforce the structured output in the strict mode. The latter guarantees that the schema will be correct (99.9%).

```suggestion
            .with_structured_output(
                MaxRecordingUniversalFilters,
                include_raw=False,
                method="json_schema",
                strict=True,
            )
```

Reply:
I've just consolidated the schema so that it's very explicitly a subset of the same types in the frontend (synced via TS->JSON->Python), so now the schema contains default Nones, which strict mode disallows. :( One other reason I'm a bit wary of strict mode, is that AFAIK Claude doesn't have it.